explore/src/main/scala/explore/tabs/ConstraintSetTabContents.scala

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2016-2021 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package explore.tabs
+
+import cats.effect.IO
+import cats.syntax.all._
+import crystal.react.implicits._
+import explore.{ AppCtx, UnderConstruction }
+import explore.common.UserPreferencesQueries._
+import explore.components.Tile
+import explore.components.ui.ExploreStyles
+import explore.implicits._
+import explore.model._
+import explore.model.reusability._
+import explore.observationtree.ConstraintSetObsList
+import explore.observationtree.ConstraintSetObsQueries._
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.component.builder.Lifecycle.ComponentDidMount
+import japgolly.scalajs.react.vdom.html_<^._
+import lucuma.core.model.{ ConstraintSet, User }
+import lucuma.ui.reusability._
+import org.scalajs.dom.window
+import react.common._
+import react.common.implicits._
+import react.draggable.Axis
+import react.resizable._
+import react.resizeDetector.ResizeDetector
+
+import scala.collection.immutable.SortedSet
+import scala.concurrent.duration._
+
+final case class ConstraintSetTabContents(
+  userId:      ViewOpt[User.Id],
+  focused:     View[Option[Focused]],
+  expandedIds: View[SortedSet[ConstraintSet.Id]],
+  size:        ResizeDetector.Dimensions
+) extends ReactProps[ConstraintSetTabContents](ConstraintSetTabContents.component) {
+  def isCsSelected: Boolean = focused.get.isDefined
```

Comment:
We might want to check if the focused element is a constraint set. Otherwise, if navigating here from another view while focused in another object type, this will be `true`.

Reply:
Hmmm. Yeah. You're right. I guess I just copied this from `TargetTabContents.scala` when I created the file. I guess we probably need to update `TargetTabContents.scala`, too?

Reply:
Yes, please.

Reply:
I've changed all of these to check the type, but we may want to think about this behavior for another PR. It seems to me like we might want Focused to be a product instead of a sum type.

Reply:
How would a product work? The idea is that in trees we can model whether, say, a target is selected or one of its observations.

Reply:
Yeah. I was looking at that code today, and the sum is needed. But, `Focused` is also being used in the `*TabContents` to determine whether to display the tree or right panel on narrow screens. This results in what seems to be odd behavior to me. For example, if you select a target in the target tab, then you'll also be displaying an empty right hand panel in constraints whether or not a constraint is selected. Alternatively, if you close a constraint and go back to the tree, you'll also go back to the tree in targets and observations whether or not you had one selected previously.

Reply:
> For example, if you select a target in the target tab, then you'll also be displaying an empty right hand panel in constraints whether or not a constraint is selected.

I might be missing something, but I think this shouldn't happen now that we are checking the type. The only case I think this can happen is when having an observation selected and switching views. I'm not sure whether this is desirable or we want to avoid it.

Reply:
You're right. I was distracted when I wrote that. That was the old behavior. The new behavior is that if you "open" a target on the target tab, you "close" any constraint that was "open" in the constraints tab, etc. Which seems like it would make switching between tabs less smooth. But, maybe it's OK.